receiver/k8slogreceiver/config.go

```diff
@@ -0,0 +1,300 @@
+	}
+	if c.StartAt != StartAtBeginning && c.StartAt != StartAtEnd {
+		err = multierr.Append(err, fmt.Errorf("invalid start_at: %s", c.StartAt))
+	}
+	return err
+}
+
+// K8sSourceConfig allows specifying how to discover containers to collect logs from.
+type K8sSourceConfig struct {
+	// Mode represents the mode of the k8slog receiver.
+	// Valid values are:
+	// - "daemonset-stdout": (default) otel is deployed as a daemonset and collects logs from stdout of containers.
+	// - "daemonset-file": otel is deployed as a daemonset and collects logs from files inside containers.
+	// - "sidecar": otel is deployed as a sidecar and collects logs from files.
+	Mode string `mapstructure:"mode"`
+
+	// NodeFromEnv represents the environment variable which contains the node name.
+	NodeFromEnv string `mapstructure:"node_from_env"`
+
+	// HostRoot represents the path which is used to mount the host's root filesystem.
+	HostRoot string `mapstructure:"host_root"`
+
+	// K8sAPI represents the configuration for the k8s API.
+	K8sAPI k8sconfig.APIConfig `mapstructure:"k8s_api"`
+
+	// RuntimeAPIs represents the configuration for the runtime APIs.
+	RuntimeAPIs []RuntimeAPIConfig `mapstructure:"runtime_apis"`
+
+	Filter []FilterConfig `mapstructure:"filter"`
+}
+
+func (c K8sSourceConfig) Validate() error {
+	var err error
+	if c.Mode != ModeDaemonSetStdout && c.Mode != ModeDaemonSetFile && c.Mode != ModeSidecar {
+		return fmt.Errorf("invalid mode %q", c.Mode)
+	}
+	if c.Mode != ModeSidecar {
+		if c.HostRoot == "" {
+			err = multierr.Append(err, fmt.Errorf("host_root must be specified when mode is %q", c.Mode))
+		}
+		err = multierr.Append(err, c.K8sAPI.Validate())
+		for _, r := range c.RuntimeAPIs {
+			err = multierr.Append(err, r.Validate())
+		}
+	}
+	return err
+}
+
+// FilterConfig allows specifying how to filter containers to collect logs from.
+// By default, all containers are collected from.
+type FilterConfig struct {
+	// Annotations represents the rules to filter containers based on pod annotations.
+	Annotations []MapFilterConfig `mapstructure:"annotations"`
+
+	// Labels represents the rules to filter containers based on pod labels.
+	Labels []MapFilterConfig `mapstructure:"labels"`
+
+	// Env represents the rules to filter containers based on pod environment variables.
+	Env []MapFilterConfig `mapstructure:"env"`
+
+	// Namespaces represents the rules to filter containers based on pod namespaces.
+	Namespaces []ValueFilterConfig `mapstructure:"namespaces"`
+
+	// Containers represents the rules to filter containers based on container names.
+	Containers []ValueFilterConfig `mapstructure:"containers"`
+
+	// Pods represents the rules to filter containers based on pod names.
+	Pods []ValueFilterConfig `mapstructure:"pods"`
+
+	// UIDs represents the rules to filter containers based on pod UIDs.
+	UIDs []ValueFilterConfig `mapstructure:"uids"`
```

Comment:
We would need to create a "Kubernetes Context" in OTTL so that we could reference stuff like `pod` and `uid`, but I can create that quickly. /cc @evan-bradley I think this would be a new context we could leverage in other k8s components based on the conversation happening in https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/25161

Reply:
I agree with using `ottl` instead of `expr`. However, I would like to keep both `ottl` and the regular filter for several reasons:
- `ottl` has a learning curve for users. I wanted to make it faster for users with simple needs.
- For pod-level conventional filters, I can reduce resource consumption by filtering out entire pods before looking up information about each container. I could also try to convert some of the simple filters to `labelSelector` or `filedSelector`, which would be more difficult with `ottl`.

Reply:
Since ottl doesn't support this context yet I'm fine without starting with it. Let's keep our configuration close to k8sattributesprocessor.